server/controllers/admin.py

```diff
@@ -637,9 +637,19 @@ def enrollment(cid):
     return render_template('staff/course/enrollment.html',
                            enrollments=students, staff=staff,
                            single_form=single_form,
+                           unenroll_form=forms.CSRFForm(),
                            courses=courses,
                            current_course=current_course)
 
+@admin.route("/course/<int:cid>/<string:user_id>/unenroll", methods=['POST'])
+@is_staff(course_arg='cid')
+def unenrollment(cid, user_id):
+    user = User.query.filter_by(id=user_id).one_or_none()
+    if user and user.is_enrolled(cid):
```

Comment:
Thanks for the comments. I didn't realize that there was a layer of caching that was involved! That is what created the 500 error. I have modified it so that when the user is unenrolled, the cache for `is_enrolled` is also cleared. Additionally, I modified the code to take advantage of the fact that `user.is_enrolled` returns the enrollment object.